USWDS - Indeterminate Checkbox: Cleanup scss and stories

mahoneycm opened the pull request:
# Summary

Runs prettier and reorganizes storybook twig files to match existing variant patterns.

## Breaking change

This is not a breaking change.

## Related issue

Resolves #5658 

## Related pull requests

#5549 
#5648 

## Preview link

[Indeterminate checkbox story →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-cleanup-scss-and-stories/?path=/story/components-form-inputs-checkbox--indeterminate)
[Indeterminate checkbox tile story →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-cleanup-scss-and-stories/?path=/story/components-form-inputs-checkbox--indeterminate-tile)

## Problem statement

Contributed work needed some minor adjustments to match USWDS patterns.

## Solution

1. Run prettier
2. Create isolated indeterminate checkbox stories
3. Remove indeterminate examples from pre-existing stories.

## Major changes

Added indeterminate storybook control and stories to display the new variant.

>[!NOTE]
> After discussion, we decided this qualified as a variant and _not_ a test.

## Testing and review

1. Confirm there are no build errors.
2. Confirm running `npm run lint:sass` does not flag any failures.
3. Confirm `npm run prettier` does not introduce any changes.
5. Existing checkbox stories matches develop.
6. Confirm test stories match the existing pattern for test features.

mejiaj commented on packages/usa-checkbox/src/usa-checkbox.stories.js:
These would be new variants, not tests. Can we update the stories so that they don't rely on test templates? This should be a new state/variant on the existing work.

mahoneycm replied:
Resolved in [34b7f68](https://github.com/uswds/uswds/pull/5650/commits/34b7f68fcfe709db30d05d025b6f932d3f0ca921)

mejiaj commented on packages/usa-checkbox/src/usa-checkbox--tile.twig:
```diff
@@ -53,32 +53,5 @@
-        {% if disabled_state == "aria-disabled" %}aria-disabled="true"{% endif %}
-      >
-      <label class="usa-checkbox__label" for="check-historical-marshall-2">Thurgood Marshall</label>
-    </div>
```
Deletion makes sense because this was added in the original work. Doesn't appear in current [develop](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/?path=/story/components-form-inputs-checkbox--aria-disabled-tile&p=all).

mejiaj replied:
Resolved in 34b7f68.

mejiaj commented on packages/usa-checkbox/src/usa-checkbox.twig:
```diff
@@ -49,33 +49,4 @@
-    >
-    <label class="usa-checkbox__label"
-    for="check-historical-marshall">Thurgood Marshall</label>
-  </div>
```
Same case as above. This was a new addition in feature.

mahoneycm replied:
Resolved in [34b7f68](https://github.com/uswds/uswds/pull/5650/commits/34b7f68fcfe709db30d05d025b6f932d3f0ca921)

mejiaj replied:
Resolved in 34b7f68.

mejiaj reviewed (changes requested):
@mahoneycm thanks for the fixes. Added a note related to treating this as a true variant instead of using test templates.

mahoneycm commented:
## Update 12/5/23

- [x]  New Issue created for this PR
- [x]  Preview links updated
- [x]  Indeterminate templates were moved out of the test directory

I included the twig files in the `src` directory. This matches the current pattern set by checkbox/radio as well as [usa-collection](https://github.com/uswds/uswds/tree/3175c0258a63e6e464ea8284bb90f452f46463a1/packages/usa-collection/src) and [usa-process-list](https://github.com/uswds/uswds/tree/3175c0258a63e6e464ea8284bb90f452f46463a1/packages/usa-process-list/src)

I noticed other components, such as [header](https://github.com/uswds/uswds/tree/3175c0258a63e6e464ea8284bb90f452f46463a1/packages/usa-header/src) or [footer](https://github.com/uswds/uswds/tree/3175c0258a63e6e464ea8284bb90f452f46463a1/packages/usa-footer/src), use a nested folder pattern but each of those templates had their own `json` content files so it seemed uneccessary for these files. Let me know if you’d prefer them to live in their own folder(s)!

amyleadem reviewed (changes requested):
Thanks for cleaning this up @mahoneycm. I added a question below about the need for breaking indeterminate styles into their own story. We don't currently do that for the `checked` state so I am curious about the reasoning. Curious to hear your thoughts.

I ran the following tests:
- [x] Confirm `npm start` runs without error
- [x] Confirm `npm run test` runs without error
- [x] Confirm the `indeterminate` state styles match the `checked` state when  `$theme-body-background-color` is set to a dark color (I tested with ”ink”)
- [x] Confirm that running prettier does not generate new fixes

mejiaj reviewed (approved):
LGTM, no issues with code.

---

Noting file size for the new assets:
1. checkbox-indeterminate.svg - `657B`
2. checkbox-indeterminate-alt.svg - `636B`

mahoneycm commented on packages/usa-checkbox/src/usa-checkbox--indeterminate-tile.twig:
Resolved in [ea6e79e](https://github.com/uswds/uswds/pull/5650/commits/ea6e79e4a2458434a81ae08ed054b78fb22228ed)

mahoneycm commented:
## Update 1/10/24

To easily compare states, I've added a checked checkbox to the indeterminate stories. I also added a storybook control to activate an indeterminate checkbox on other stories as well. In my opinion, this follows similar patterns for other states such as the disabled state for this component.

I also removed the indeterminate twig files in favor of using the standard template with the indeterminate state permanently set to true. 

PR summary also adjusted to remove "test" references and the decision to treat this as a variant.

amyleadem reviewed (approved):
The update looks great! I was able to easily perform visual comparisons in the stories. Thanks for updating that. 

@mejiaj, any objections to this updated approach?

mejiaj commented:
LGTM, thank you both
